app/controllers/v1/sessions_controller.rb

```diff
@@ -101,10 +100,12 @@ def authenticate
     end
 
     def user_login(saml_response)
-      user_session_form = UserSessionForm.new(saml_response)
-      unless user_session_form.valid?
-        login_stats(:failure, saml_response, user_session_form)
-        raise_saml_error(user_session_form)
+      begin
+        user_session_form = UserSessionForm.new(saml_response)
+        raise_saml_error(user_session_form) unless user_session_form.valid?
+      rescue SAML::SAMLError => e
```

Comment:
Exactly which error are we accounting for here? Is this to catch `SAML::UserAttributeError`s from `UserSessionForm#normalize_saml`?
Are there any other `SAML::SAMLError`s that might occur?

Reply:
yeah thats a good point, it should really only be `SAML::UserAttributeError` that could be raised on line 104.  I'll update the `rescue`